packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -186,9 +196,74 @@ export class AssetCode extends Code {
       this.asset!.addResourceMetadata(resource, 'Code');
     }
 
-    return  {
+    return {
       s3Bucket: this.asset!.s3BucketName,
       s3Key: this.asset!.s3ObjectKey
     };
   }
 }
+
+export interface CfnParametersCodeProps {
+  readonly bucketNameParam?: cdk.CfnParameter;
+
+  readonly objectKeyParam?: cdk.CfnParameter;
+}
+
+/**
+ * Lambda code defined using CloudFormation parameters.
+ * Useful when you don't have access to the code of your Lambda from your CDK code,
+ * so you can't use Assets,
+ * and you want to deploy the Lambda in a CodePipeline, using CloudFormation -
+ * you can fill the parameters using the
+ * `codepipeline.Artifact#fillParameters(CfnParametersCode)` method.
+ */
+export class CfnParametersCode extends Code {
+  public readonly isInline = false;
+  private _bucketNameParam?: cdk.CfnParameter;
+  private _objectKeyParam?: cdk.CfnParameter;
+
+  constructor(props: CfnParametersCodeProps = {}) {
+    super();
+
+    this._bucketNameParam = props.bucketNameParam;
+    this._objectKeyParam = props.objectKeyParam;
+  }
+
+  public bind(construct: cdk.Construct) {
+    if (!this._bucketNameParam) {
+      this._bucketNameParam = new cdk.CfnParameter(construct, 'LambdaSourceBucketNameParameter', {
+        type: 'String',
+      });
+    }
+
+    if (!this._objectKeyParam) {
+      this._objectKeyParam = new cdk.CfnParameter(construct, 'LambdaSourceObjectKeyParameter', {
+        type: 'String',
+      });
+    }
+  }
+
+  /** @internal */
+  public _toJSON(_?: cdk.CfnResource): CfnFunction.CodeProperty {
+    return {
+      s3Bucket: this._bucketNameParam!.stringValue,
```

Comment:
No !, Use the getters

Reply:
Can't - the getters only return the names of the parameters, while I need their values here.